src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -920,6 +929,24 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		this._connectionUrisToDispose = [];
 	}
 
+	// Track time it takes to render all markdown cells
+	private trackMarkdownTelemetry(cellContent: nb.ICellContents, cellModel: ICellModel): void {
+		if (cellContent && cellContent.cell_type === CellTypes.Markdown) {
+			this._textCellsLoading++;
+		}
+		this._register(cellModel.onLoaded((e) => {
+			this._textCellsLoading--;
```

Comment:
You should only add this callback if it's a text cell - otherwise non-text cells will incorrectly decrement the _textCellsLoading count

Reply:
thanks. I'd like to do a similar thing for code cells, so made the check inside the callback.